torch/_export/verifier.py

```diff
@@ -31,6 +31,35 @@ def _check_has_fake_tensor(node: torch.fx.Node) -> None:
     return _check_val(node)
 
 
+def check_nn_module_stack(graph_module: torch.fx.GraphModule) -> None:
+    '''
+    Perform nn_module_stack checks on the graph.
+    Current constraints:
+        For the top level graph:
+        - populated for 'call_method', 'call_function', 'get_attr', 'call_module'
+        - None for 'placeholder', 'output'
+        For submodule graphs:
+        - None for 'placeholder', output'
+
+    TODO(pianpwk): extend this check to get_attr & call_module nodes.
+    TODO(pianpwk): make this a consistent node-level check once nn_module_stack is populated for cond submodules.
+    '''
+    # Check top-level graph for all nodes, submodule graphs for placeholder & output nodes
+    for i, mod in enumerate(graph_module.modules()):
+        for node in mod.graph.nodes:
+            if node.op in ['call_method', 'call_function', 'get_attr', 'call_module']:
+                if i == 0:
```

Comment:
hmm, I'm not sure if the ordering is consistent in the `.modules()` calls, where i=0 is always the toplevel module. Maybe it's better if you first run a check on `graph_module` call_function nodes, and then do loop over all the modules to check for the placeholder/output nodes.

Reply:
Good point, thank you! Just changed :)